dart/lib/src/environment_variables.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:meta/meta.dart';
+
+import 'sentry_options.dart';
+
+/// This method reads available environment variables and uses them
+/// accordingly.
+/// To see which environment variables are available, see [EnvironmentVariables]
+///
+/// The precendence of these options are tricky,
+/// see https://docs.sentry.io/platforms/dart/configuration/options/
+/// and https://github.com/getsentry/sentry-dart/issues/306
+@internal
+void setEnvironmentVariables(SentryOptions options, EnvironmentVariables vars) {
+  // options has precendence over vars
+  options.dsn = options.dsn ?? vars.dsn;
+
+  var environment = options.platformChecker.environment;
+  options.environment = options.environment ?? environment;
+
+  // vars has precedence over options
+  options.environment = vars.environment ?? options.environment;
+
+  // vars has precedence over options
+  options.release = vars.release ?? options.release;
+  options.dist = vars.dist ?? options.dist;
```

Comment:
after reading the docs again I guess I've misunderstood it and we should fix it, please help me if I am wrong.

options should always have precedence over vars, the user always has the chance to overwrite any value so it must take precedence.

* use options.dsn and if not, use vars.dsn
* use options.environment and if not, use vars.environment and if not, use options.platformChecker.environment
* use options.release and if not, vars.release and if not, package_info
* use options.dist and if not, vars.dist

`OptionsConfiguration` always run last so the user always should have the chance to overwrite any value, the gotcha here is for `release` as the integration reads the package_info after that

sorry for the confusion.

Reply:
So I could pass `EnvironmentVariables` as a constructor parameter to `SentryOptions` and set them.
The user is then free to overwrite them in the `OptionsConfiguration`. Then `EnvironmentVariables` does not have to be a property of `SentryOptions`.

Reply:
I dont think so, users never creates their own `SentryOptions`, but always mutate the SDKs `SentryOptions`, the reason is, we always have the chance to set default values before the user has the chance to mutate it, so I'd be in favor of being a property to `SentryOptions` that we read from, like all the other internal properties, users dont have to interact with `EnvironmentVariables`